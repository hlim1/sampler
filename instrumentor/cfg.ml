open Cil
  

type cfg = stmt * stmt list

let cfg func =
  prepareCFG func;
  let stmts = computeCFGInfo func false in
  let root = List.hd func.sbody.bstmts in

  if root.sid != 0 then
    ignore(bug "first statement in function body is not root of CFG");

  (root, stmts)
