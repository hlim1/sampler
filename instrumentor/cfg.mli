open Cil


val hasDefault : stmt list -> bool

val build : fundec -> unit
