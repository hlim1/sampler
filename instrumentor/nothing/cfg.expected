object cfg.o
	unit cfg.c
		function conditional at cfg.c:14
			node 0 at cfg.c:16
				successors: 2 1
			node 1 at cfg.c:17
				successors: 3
				callees: loopy
			node 2 at cfg.c:19
				successors: 3
				callees: conditional
			node 3 at cfg.c:20
		function loopy at cfg.c:7
			node 0 at cfg.c:9
				successors: 1
			node 1 at cfg.c:9
				successors: 2
				callees: (opaque)
			node 2 at cfg.c:9
				successors: 4 3
			node 3 at (unknown):0
				successors: 5
			node 4 at cfg.c:9
				successors: 6
			node 5 at cfg.c:10
				successors: 0
			node 6 at cfg.c:11
